### PR TITLE
Fix no connection bug #3

### LINKED ISSUE #3: [BUG] Fails when no connection is configured
### Formula commit hash / release tag

c736df16da3df80e8a5fee46fa8074cae09f19cb (before semantic-release implementation)

### Versions reports (master & minion)

not applicable

### Pillar / config used

pillar used:

```
config:
        ----------
        strictcrlpolicy:
            no
        uniqueids:
            yes
    conn:
        None
    secrets:
        None
```
---

### Describe the bug

When applying `strongswan` with pillars above, code fails with a message about `strongswan.config' failed: Jinja variable 'None' has no attribute 'items'`

### Steps to reproduce the bug
Install formula, initialize pillars as above with nothing in `conn` or `secrets`, run the strongswan state `state.sls strongswan`

### Expected behaviour

It should succeed (install packages, initialize default config) without creating any connections and secrets.

### Attempts to fix the bug

working on it

## Additional context

Secrets management may have the same problem.

### PR DESCRIPTION
**Pull request by daks**
# **WIP, DO NOT MERGE**

### PR progress checklist (to be filled in by reviewers)
- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type
- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [X] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type
- [ ] `[docs]`     Documentation changes
- [X] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?
No.

### Related issues and/or pull requests

Fix #3 

### Describe the changes you're proposing

- Add new Inspec test suite for a no configuration setup
- fix no connection bug https://github.com/saltstack-formulas/strongswan-formula/issues/3
- remove duplicate test on `/etc/ipsec.secrets`

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist
- [X] Included in Kitchen (i.e. under `state_top`).
- [X] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

File: strongswan/config.sls [hunk 1]
```diff
@@ -12,13 +12,15 @@
...
 {%- set conn_default = {} %}
 
-{%- for k, v in conn.items() %}
-  {%- if k == '_default' %}
-    {%- do conn_default.update(v) %}
-  {% else %}
-    {%- do connections.update({k:v}) %}
-  {%- endif %}
-{%- endfor %}
+{%- if conn %}
```
Comment (javierbertoli) [inline strongswan/config.sls:15]: @daks, I took a quick glimpse of the PR and fail to see why you need to add this conditional here, as [the default value for `conn`, if not set is an empty value](https://github.com/saltstack-formulas/strongswan-formula/blob/e99c764c9f529425e677ce22cb83ec6bd5a25f1b/strongswan/config.sls#L11) and, therefore, this loops is skipped. Am I missing something?

Also, I think that probably is better to set a default value for `conn` in `defaults.yaml` and change that same line I mention above accordingly, wdyt?

**Review by javierbertoli — COMMENTED** [review]
@daks, added a little comment. The rest of the PR, is great! :smile:

File: strongswan/config.sls (hunk 1 above)
Comment (daks) [reply strongswan/config.sls:15]: I was surprised by this bug too thinking that `{%- set conn = salt['pillar.get']('strongswan:conn', {}) %}` was sufficient. But it's not :) The first commit implements a test which proves this bug.

I prefer to think that `defaults.yaml` is used for 'core' parameters to configure the formula (like config directory, file names...) and the distinction between OS and releases.
I don't think that default parameters to be passed to the managed software itself should be put in `defaults.yaml`.

Moreover the `conn` configuration is what the user need to define, it has no sense to have a 'default' value which be useless for anyone.

**Comment by daks** [comment]
no bug, closing.

**Pull request closed without merging**